File: cs373/1/java/RunCollatz.java
// --------------------------------
// projects/collatz/RunCollatz.java
// Copyright (C) 2010
// Glenn P. Downing
// --------------------------------

/*
To run the program:
    % javac -Xlint RunCollatz.java
    % java  -ea    RunCollatz < RunCollatz.in > RunCollatz.out

To document the program:
    % javadoc -d html -private *.java
*/

import java.io.IOException;
import java.io.PrintWriter;
import java.io.Writer;
import java.util.Scanner;

final class RunCollatz
{
	public static void main (String[] args) throws IOException
	{
		final Scanner s;
		final Writer w;
		final int[] a;

		s = new Scanner(System.in);
		w = new PrintWriter(System.out);
		a = new int[2];

		while (Collatz.read(s, a))
		{
			final int v = Collatz.eval(a);
			Collatz.print(w, a, v);
		}
	}
}

File: cs373/1/python/RunCollatz.py
#!/usr/bin/env python

# -------------------------------------
# projects/python/collatz/RunCollatz.py
# Copyright (C) 2010
# Glenn P. Downing
# -------------------------------------

# To run the program
#     RunCollatz.py < RunCollatz.in > RunCollatz.out

# To document the program
#     pydoc -w Collatz RunCollatz TestCollatz

from Collatz import collatz_read, collatz_eval, collatz_print

# ------
# Reader
# ------

class Reader (object) :
    def read (self) :
        return raw_input()

# ------
# Writer
# ------

class Writer (object) :
    def write (self, a, v) :
        for i in a :
            print i,
        print v

# ----
# main
# ----

def main () :
    """
    runs the program
    """
    a = []
    while collatz_read(Reader(), a) :
        v = collatz_eval(a)
        collatz_print(Writer(), a, v)

if __name__ == "__main__" :
    main()